src/jdbc/cubrid/jdbc/net/BrokerHandler.java

```diff
@@ -17,175 +17,238 @@
-		if (timeout <= 0) {
-		    throw new UJciException(UErrorCode.ER_TIMEOUT);
-		}
-		toBroker.connect(brokerAddress, timeout);
-	    }
-
-	    toBroker.setKeepAlive(true);
-	    return toBroker;
-	} catch (SocketTimeoutException e) {
-	    if (toBroker != null) {
-		toBroker.close();
-	    }
-	    throw new UJciException(UErrorCode.ER_TIMEOUT, e);
-	} catch (IOException e) {
-	    if (toBroker != null) {
-		toBroker.close();
-	    }
-	    throw new UJciException(UErrorCode.ER_CONNECTION, e);
-	} finally {
-	}
+            throws IOException, UJciException {
+        Socket toBroker = null;
+        UTimedDataInputStream in = null;
+        DataOutputStream out = null;
+        long begin = System.currentTimeMillis();
+
+        try {
+          toBroker = new Socket();
+          InetSocketAddress brokerAddress = new InetSocketAddress(ip, port);
+          if (timeout <= 0) {
+            toBroker.connect(brokerAddress);
+          }
+          else {
+            toBroker.connect(brokerAddress, timeout);
+            timeout -= (System.currentTimeMillis() - begin);
+            if (timeout <= 0) {
+              toBroker.close();
+              throw new UJciException(UErrorCode.ER_TIMEOUT);
+            }
+          }
+
+          toBroker.setSoTimeout(TIMEOUT_UNIT);
+          toBroker.setKeepAlive(true);
+          in = new UTimedDataInputStream(toBroker.getInputStream(), ip, port, timeout);
+          out = new DataOutputStream(toBroker.getOutputStream());
+          out.write(UConnection.driverInfo);
+          out.flush();
+          int code = in.readInt();
+          if (code < 0) {
+          // in here, all errors are sent by only a broker
+          // the error greater than -10000 is sent by old broker
+            if (code > -10000) {
+              code -= 9000;
+            }
+            throw new UJciException(code);
+          }
+          else if (code == 0) {
+            return toBroker;
+          }
+
+          // if (code > 0) { only windows }
+          toBroker.setSoLinger(true, 0);
+          toBroker.close();
+
+          toBroker = new Socket(); // need instantiation
+          brokerAddress = new InetSocketAddress(ip, code);
+          if (timeout <= 0) {
+            toBroker.connect(brokerAddress);
+          }
+          else {
+            timeout -= (System.currentTimeMillis() - begin);
+            if (timeout <= 0) {
+              throw new UJciException(UErrorCode.ER_TIMEOUT);
+            }
+            toBroker.connect(brokerAddress, timeout);
+          }
+
+          toBroker.setKeepAlive(true);
+          return toBroker;
+        } catch (SocketTimeoutException e) {
+          if (toBroker != null) {
+          toBroker.close();
+        }
+        throw new UJciException(UErrorCode.ER_TIMEOUT, e);
+        } catch (IOException e) {
+          if (toBroker != null) {
+          toBroker.close();
+        }
+        throw new UJciException(UErrorCode.ER_CONNECTION, e);
+        } finally {
+        }
```

Comment:
When checked by eye, the above source block doesn't seem to be modified.
In my opinion, it should not be modified for indentation purposes.

Reply:
You are right, but the coding style is not defined finely for JAVA so I modified the code partially for readability.
In the future, all of the Java code will be rewritten by the coding style we define.

Reply:
I found _createJciException()_ method which dumps exception to the logger and returns the UJciException. I think using this method seems better.